python/cm/models.py

```diff
@@ -285,17 +285,20 @@ class ConfigLog(ADCMModel):
     def save(self, *args, **kwargs):
         """Saving config and updating config groups"""
 
-        def update(origin, renovator):
+        def update(origin, renovator, group_keys):
             """
             Updating the original dictionary with a check for the presence of keys in the original
             """
-            for key, value in renovator.items():
-                if key not in origin:
-                    continue
-                if isinstance(value, Mapping):
-                    origin[key] = update(origin.get(key, {}), value)
+            if origin is None:
+                origin = {}
+            for key, value in group_keys.items():
+                if isinstance(value, Mapping) and key in renovator:
+                    origin[key] = update(
+                        origin.get(key, {}), renovator.get(key, {}), group_keys[key]
+                    )
                 else:
-                    origin[key] = value
+                    if value and key in renovator:
+                        origin[key] = renovator[key]
```

Comment:
Seems like it has side-effect contradicting with line 293 - sometime you change object in place, sometime return new one

Reply:
fixed